autorest/codegen/serializers/__init__.py

```diff
@@ -72,6 +98,26 @@ def serialize(self, code_model: CodeModel) -> None:
             self._serialize_and_write_models_folder(code_model=code_model, env=env, namespace_path=namespace_path)
 
 
+        if code_model.options["package_mode"] == 'dataplane':
```

Comment:
I think the scope of this PR is also too big: the issue we're trying to tackle here is only for the setup.py https://github.com/Azure/autorest.python/issues/1053.

We want to make sure the PRs are more atomic bc we don't necessarily want to add all of this at the same time, better to keep it smaller so it's easier to see the changes as they come

Reply:
same discusstion with: https://github.com/Azure/autorest.python/pull/1112/files#r792351788